hv/crates/hv-cell/src/lib.rs

```diff
@@ -178,6 +179,75 @@ impl<T: ?Sized> ArcCell<T> {
     pub fn from_inner(inner: Arc<AtomicRefCell<T>>) -> Self {
         Self { inner }
     }
+
+    /// Construct a `WeakCell<T>`
+    pub fn downgrade(this: &Self) -> WeakCell<T> {
+        WeakCell {
+            inner: Arc::downgrade(&this.inner),
+        }
+    }
+}
+
+/// An analogue to `Arc<T>`'s `Weak<T>` type, normally constructed via `ArcCell::downgrade(this: &T)` or `WeakCell::new()`
+pub struct WeakCell<T: ?Sized> {
+    inner: Weak<AtomicRefCell<T>>,
+}
+
+impl<T> Default for WeakCell<T> {
+    fn default() -> Self {
+        WeakCell::new()
+    }
+}
+
+impl<T> WeakCell<T> {
+    #[inline]
+    /// Construct a new `WeakCell<T>`
+    pub fn new() -> Self {
+        WeakCell { inner: Weak::new() }
+    }
+
+    /// Retrieve the inner field of the `WeakCell<T>`
+    pub fn as_inner(&self) -> &Weak<AtomicRefCell<T>> {
+        &self.inner
+    }
+
+    /// Unsafely try to borrow the underlying `T` (note that this could panic either due to a `borrow`
```

Comment:
A panic is not unsafe; instead, it should be noted that this method will panic when the borrow is not possible